pkg/localmetrics/localmetrics.go

```diff
@@ -15,22 +15,81 @@
 package localmetrics
 
 import (
+	"context"
+	"net/http"
+	"time"
+
+	pdApi "github.com/PagerDuty/go-pagerduty"
+	corev1 "k8s.io/api/core/v1"
+
 	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var (
-	metricPlaceholder = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "placeholder",
-		Help: "Placeholder",
+	MetricPagerDutyCreateFailure = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "metricPagerDutyCreateFailure",
+		Help: "metricPagerDutyCreateFailure",
+	}, []string{"name", "clusterdeployment_name"})
+
+	MetricPagerDutyDeleteFailure = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "metricPagerDutyDeleteFailure",
+		Help: "metricPagerDutyDeleteFailure",
+	}, []string{"name", "clusterdeployment_name"})
+
+	MetricPagerDutyHeartbeat = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "metricPagerDutyHeartbeat",
+		Help: "Metric for heartbeating the Dead Man's Snitch api",
 	}, []string{"name"})
 
 	MetricsList = []prometheus.Collector{
-		metricPlaceholder,
+		MetricPagerDutyCreateFailure,
+		MetricPagerDutyDeleteFailure,
+		MetricPagerDutyHeartbeat,
 	}
 )
 
-// UpdatePlaceholderGauge ...
-func UpdatePlaceholderGauge() {
+// UpdateAPIMetrics updates all API endpoint metrics ever 5 minutes
+func UpdateAPIMetrics(client client.Client) error {
+
+	pdAPISecret := &corev1.Secret{}
+	err := client.Get(context.TODO(), types.NamespacedName{Namespace: "pagerduty-operator", Name: "pagerduty-api-key"}, pdAPISecret)
+	if err != nil {
+		return err
+	}
+
+	APIkey := string(pdAPISecret.Data["PAGERDUTY_API_KEY"])
+
+	d := time.Tick(5 * time.Minute)
+	for range d {
+		UpdateMetricPagerDutyHeartbeatGauge(APIkey)
+	}
+}
+
+// UpdateMetricPagerDutyCreateFailure updates guage to 1 when creation fails
+func UpdateMetricPagerDutyCreateFailure(x int, cd string) {
+	MetricPagerDutyCreateFailure.With(prometheus.Labels{"name": "pagerduty-operator", "clusterdeployment_name": cd}).Set(float64(x))
+}
+
+// UpdateMetricPagerDutyDeleteFailure updates guage to 1 when deletion fails
+func UpdateMetricPagerDutyDeleteFailure(x int, cd string) {
+	MetricPagerDutyDeleteFailure.With(prometheus.Labels{"name": "pagerduty-operator", "clusterdeployment_name": cd}).Set(float64(x))
+}
+
+// UpdateMetricPagerDutyHeartbeatGauge curls the PD API, updates the gauge to 1 when successful.
+func UpdateMetricPagerDutyHeartbeatGauge(APIKey string) {
+
+	_, err := http.NewRequest("GET", "api.pagerduty.com", nil)
+	if err != nil {
```

Comment:
Should we log an error here? /cc @fahlmant

Reply:
+1

Reply:
Done